screeps-start.js

```diff
@@ -81,6 +91,74 @@ const installPackages = () => {
   }
 
   console.log("Done updating");
+}
+
+const updatePackages = (doUpdate) => {
+  const mods = config.mods || [];
+  const bots = config.bots || {};
+
+  const modsPackage = loadPackage(ModsDir);
+  const dependencies = modsPackage.dependencies || {};
+
+  // Calculate package diff
+  const configuredPackages = [...mods, ...Object.values(bots)];
+
+  const packagedMods = configuredPackages.filter(
+    (pkg) =>
+      Object.entries(dependencies).some((dependency) =>
+        isDependency(pkg, dependency),
+      ),
+  ).map((pkg) => parseVersionSpec(pkg));
+
+  let outdated = {};
+  const outdatedFile = path.resolve(ModsDir, "outdated.json");
+  try {
+    // `npm outdated --json` returns 1 if there are outdated packages,
+    // which causes `execSync` to throw an error.
+    execSync("npm outdated --json > outdated.json || true", {
+      cwd: ModsDir,
+      stdio: "inherit",
+      encoding: "utf8",
+      shell: true,
+    })
+    const output = fs.readFileSync(outdatedFile)
+    outdated = JSON.parse(output);
+  } catch {
+  } finally {
+    try {
+      fs.unlinkSync(outdatedFile);
+    } catch {
+    }
+  }
+
+  const versionSpecs = [];
+  for (const [mod, info] of Object.entries(outdated)) {
+    const [name, version] = packagedMods.find(([pkg]) => mod === pkg) || [];
+    if (!name) continue;
+    if (version !== "latest") {
+      console.log(`package ${name} is pinned to version ${version}, ignoring`);
+      continue;
+    }
+    versionSpecs.push(`${mod}@${info.latest}`);
+  }
```

Comment:
I would probably lean towards just checking your `configuredMods` without the parsing. See if `version == "latest"`, then use `info.latest` else use `info.wanted`?
That may simplify a bit, and avoid some annoying edge-cases in `parseVersionSpecs`?

Reply:
I'm not sure I understand… What sort of edge cases? `info.wanted` is always gonna be the currently installed version, because all the dependencies are added as "exact matches" (with `-E`), so it wouldn't update anything.